x-pack/test/apm_api_integration/tests/service_groups/service_group_count.spec.ts

```diff
@@ -0,0 +1,206 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+import { apm, timerange } from '@kbn/apm-synthtrace';
+import expect from '@kbn/expect';
+import { ApmRuleType } from '@kbn/apm-plugin/common/rules/apm_rule_types';
+import { FtrProviderContext } from '../../common/ftr_provider_context';
+import { waitForActiveAlert } from './wait_for_active_alert';
+
+export default function ApiTest({ getService }: FtrProviderContext) {
+  const registry = getService('registry');
+  const apmApiClient = getService('apmApiClient');
+  const supertest = getService('supertest');
+  const synthtraceEsClient = getService('synthtraceEsClient');
+  const esDeleteAllIndices = getService('esDeleteAllIndices');
+  const esClient = getService('es');
+  const log = getService('log');
+  const start = Date.now() - 24 * 60 * 60 * 1000;
+  const end = Date.now();
+
+  async function callApi() {
+    return apmApiClient.readUser({
+      endpoint: 'GET /internal/apm/service-group/counts',
+    });
+  }
+
+  async function saveServiceGroup({
+    serviceGroupId,
+    groupName,
+    kuery,
+    description,
+    color,
+  }: {
+    serviceGroupId?: string;
+    groupName: string;
+    kuery: string;
+    description?: string;
+    color?: string;
+  }) {
+    return apmApiClient.writeUser({
+      endpoint: 'POST /internal/apm/service-group',
+      params: {
+        query: {
+          serviceGroupId,
+        },
+        body: {
+          groupName,
+          kuery,
+          description,
+          color,
+        },
+      },
+    });
+  }
+
+  type SavedObjectsFindResults = Array<{
+    id: string;
+    type: string;
+  }>;
+
+  async function deleteServiceGroups() {
+    const response = await supertest
+      .get('/api/saved_objects/_find?type=apm-service-group')
+      .set('kbn-xsrf', 'true');
+    const savedObjects: SavedObjectsFindResults = response.body.saved_objects;
+    const bulkDeleteBody = savedObjects.map(({ id, type }) => ({ id, type }));
+    return supertest
+      .post(`/api/saved_objects/_bulk_delete?force=true`)
+      .set('kbn-xsrf', 'true')
+      .send(bulkDeleteBody);
```

Comment:
Unless it is much slower, we should use our own APIs. 

To retrieve all service groups:
```
GET /internal/apm/service-groups
```

Then, to delete a service group:
```
DELETE /internal/apm/service-group
```

Reply:
Yes, makes sense, I used our own for the e2e tests

Reply:
the delete is faster and works better to call the saved_objects

Reply:
Yes, I saw :)